File: 204. Count Primes.cpp
// timeout

class Solution {
public:
    int countPrimes(int n) {
        if(n == 0 || n == 1) return 0;
        vector <int> T (n, 1);
        T[0] = 0, T[1] = 0;
        for (int j = 2; j*j <= n; j++)
        {
            for (int i = j+1; i < n; i++)
            {
                if (i % j == 0) T[i] = 0;
            }
        }
        int res = 0;
        for (int i = 0; i < n; i++)
            res += T[i];
        return res;
    }
};

File: 204. Count Primes.py
# timeout

class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0 or n == 1:
            return 0
        T = [1 for i in xrange(n)]
        T[0] = 0
        T[1] = 0
        
        for j in xrange(2, n):
            if j*j <= n:
                for i in xrange(j+1, n):
                    if i % j == 0:
                        T[i] = 0
            else:
                break
        return sum(T)
